soc/silabs/silabs_siwx917/siwg917/nwp_init.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+
+#include "sl_wifi.h"
+#include "sl_wifi_callback_framework.h"
+#if IS_ENABLED(CONFIG_BT_SIWX917)
+#include "rsi_ble_common_config.h"
+#endif
+
+/* TODO SILABS ZEPHYR: Check how to add constants to the vector table */
+#define NWP_ADD_TO_VECTOR_TABLE(irq, flags, expression, arg, name)                                 \
+	static Z_DECL_ALIGN(struct _isr_list) Z_GENERIC_SECTION(.intList)                          \
+		__used __isr_##name = {irq, flags, expression, arg}
+
+#if IS_ENABLED(CONFIG_WIFI_SIWX917) && IS_ENABLED(CONFIG_BT_SIWX917)
+#define NWP_INIT_COEX_MODE SL_SI91X_WLAN_BLE_MODE
+#elif IS_ENABLED(CONFIG_WIFI_SIWX917)
+#define NWP_INIT_COEX_MODE SL_SI91X_WLAN_ONLY_MODE
+#elif IS_ENABLED(CONFIG_BT_SIWX917)
+#define NWP_INIT_COEX_MODE SL_SI91X_BLE_MODE
+#else
+#error "Not supported configuration!"
+#endif
```

Comment:
If your idea is to report configuration error, you can write 
```
BUILD_ASSERT(IS_ENABLED(CONFIG_WIFI_SIWX917) || IS_ENABLED(CONFIG_BT_SIWX917));
```

Reply:
I prefer to close the if-elseif-elseif... conditions with an else than an "independent" build assert.